persistence/remote/src/main/java/org/infinispan/persistence/remote/upgrade/MigrationTask.java

```diff
@@ -112,18 +111,15 @@ private void migrateEntriesWithMetadata(RemoteCache<Object, Object> sourceCache,
                int lifespan = metadataValue.getLifespan();
                int maxIdle = metadataValue.getMaxIdle();
                long version = metadataValue.getVersion();
-               long created = metadataValue.getCreated();
-               long lastUsed = metadataValue.getLastUsed();
                Metadata metadata = new EmbeddedMetadata.Builder()
                      .version(new NumericVersion(version))
                      .lifespan(lifespan, TimeUnit.SECONDS)
                      .maxIdle(maxIdle, TimeUnit.SECONDS)
                      .build();
-               InternalMetadataImpl internalMetadata = new InternalMetadataImpl(metadata, created, lastUsed);
                executorService.submit(() -> {
                   Object key = entry.getKey();
                   if (!deletedKeys.contains(new WrappedByteArray((byte[]) key))) {
-                     cache.getAdvancedCache().withFlags(Flag.SKIP_CACHE_LOAD, Flag.ROLLING_UPGRADE).putIfAbsent(entry.getKey(), entry.getValue().getValue(), internalMetadata);
+                     cache.getAdvancedCache().withFlags(Flag.SKIP_CACHE_LOAD, Flag.ROLLING_UPGRADE).putIfAbsent(entry.getKey(), entry.getValue().getValue(), metadata);
```

Comment:
Now that the updated ISPN-4075 is in, we can do `new InternalMetadataImpl(metadata, created, lastUsed)` and it should preserve the timestamps. Maybe it would be better in another PR/JIRA though.